crates/primitives/starknet/src/transaction/execution.rs

```diff
@@ -575,6 +601,82 @@ mod simulate_tx_offset {
         );
     }
 
-    // TODO: add test that check that each get_account_transaction_context impl correctly uses this
-    // offset
+    #[test]
+    fn transaction_version_trait_correctly_applies_simulate_tx_version_offset() {
+        let v0 = TransactionVersion(StarkFelt::from(0u8));
+        assert_eq!(v0.clone().apply_simulate_tx_version_offset(), apply_simulate_tx_version_offset(v0.clone(), true));
```

Comment:
This just test that the rust syntax `variable.function()` does the same this as the `function(variable)` one. 

This is not a test we want to keep